### PR TITLE
Allow copying invalid iterators

### PR DESCRIPTION
**Pull request by bangerth**
Over at https://github.com/geodynamics/aspect/pull/1904#pullrequestreview-57643369 , @jdannberg figured out that copying invalid DoF iterators yields an exception. This makes no sense, so make sure it is actually possible.

### CODE REVIEW

**Comment by drwells** [comment]
Does `hp::DoFHandler` have the same issue?

**Comment by bangerth** [comment]
It goes through the same code path (the DoF accessors have the DoF Handler type as a template argument).

**Comment by drwells** [comment]
Good point.

**Comment by drwells** [comment]
wait, it looks like tjhei-alpha is back.

/dont-run-tests

**Comment by tjhei** [comment]
> wait, it looks like tjhei-alpha is back.

It was never gone, but it stopped marking things for a few hours. I improved the monitoring and it shouldn't happen again.

> /dont-run-tests

Not sure what you are trying to accomplish but this has no effect.

File: doc/news/changes/minor/20170823Bangerth [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Fixed: It turns out that it was not possible to copy an invalid
+operator, i.e., the following code would yield an exception:
+@code
+  typename DoFHandler<dim>::active_cell_iterator invalid_1;
+  typename DoFHandler<dim>::active_cell_iterator invalid_2;
+
+  invalid_1 = invalid_2;  // resulted in an error
+@endcode
+This made no sense, and has now been fixed
```
Comment (tjhei) [inline doc/news/changes/minor/20170823Bangerth]: period.

**Comment by tjhei** [comment]
> It goes through the same code path (the DoF accessors have the DoF Handler type as a template argument).

Can we get a test for the hp case, then please?

**Comment by drwells** [comment]
I tried to turn off the `gcc-serial` tester.

Lets wait on the changes @tjhei requested.

**Comment by bangerth** [comment]
I added a second test for the hp case and fixed the missing period.

**Comment by tjhei** [comment]
> I tried to turn off the gcc-serial tester.

The tester keeps running if it already started. If it hasn't started yet you need to edit/remove "/run-tests" from all comments in this PR. There is currently no other way to control it.

**Pull request merged**